core/src/com/unciv/logic/civilization/managers/ThreatManager.kt

```diff
@@ -0,0 +1,128 @@
+package com.unciv.logic.civilization.managers
+
+import com.unciv.logic.civilization.Civilization
+import com.unciv.logic.map.mapunit.MapUnit
+import com.unciv.logic.map.tile.Tile
+
+class ThreatManager(val civInfo: Civilization) {
+
+    data class ClosestEnemyTileData(
+        // The farthest radius in which we have checked all the tiles for enemies
+        // A value of 2 means there are no enemies in a radius of 2
+        var distanceSearched: Int,
+        // It is guaranteed that there is no enemy within a radius of D-1
+        // The enemy that we saw might have been killed 
+        // so we have to check the tileWithEnemy to see if we need to search again
+        var distanceToClosestEnemy: Int? = null,
+        // Stores the location of the enemy that we saw
+        // This allows us to quickly check if they are still alive
+        // and if we should search farther
+        var tileWithEnemy: Tile? = null
+    )
+
+    private val distanceToClosestEnemyTiles = HashMap<Tile, ClosestEnemyTileData>()
```

Comment:
I didn't find civInfo.threatManager.clearThreatData() necessary in testing. I think the Civ and, therefore the ThreatManager would be regenerated either after each Civ's turn or at the start of the player's turn when we reload the world. I added in the clearThreatData() just to be sure. So it would be helpful if anyone has more insight on this.